main.cpp

```diff
@@ -1089,12 +1114,28 @@ int main(int argc, char ** argv) {
     // the first thing we will do is to output the prompt, so set color accordingly
     set_console_state(CONSOLE_STATE_PROMPT);
 
+    if (params.embedding){
+        embd = embd_inp;
+        if (embd.size() > 0) {
+            const int64_t t_start_us = ggml_time_us(); 
+            if (!llama_eval(model, params.n_threads, n_past, embd, logits, mem_per_token, true)) {
```

Comment:
pass in the `params.embedding` instead, which presumably contains the same information about whether or not to use the embeddings as input